components/bellsound/ConvertStep.tsx

```diff
@@ -0,0 +1,113 @@
+"use client"
+
+import { useRef, useState } from "react"
+import { CommonProps } from "./BellSoundWalkthrough"
+import { FFmpeg } from "@ffmpeg/ffmpeg"
+import { fetchFile, toBlobURL } from "@ffmpeg/util"
+import WalkthroughButton from "./WalkthroughButton"
+
+export default function ConvertStep({ onDismiss, selectedFile, onConversionCompleted }: CommonProps & {
+    selectedFile: File,
+    onConversionCompleted: (convertedFile: Uint8Array) => void,
+}) {
+    const [ffmpegLog, setFfmpegLog] = useState<string>("")
+    const [showLog, setShowLog] = useState<boolean>(false)
+    const [converting, setConverting] = useState<boolean>(false)
+    const ffmpegRef = useRef<FFmpeg>(new FFmpeg())
+
+    const log = (message: string) => {
+        console.log(`[ConvertStep] ${message}`)
+        setFfmpegLog((prev) => prev + message + "\n")
+    }
+
+    const loadFfmpeg = async () => {
+        const baseURL = "https://unpkg.com/@ffmpeg/core@0.12.2/dist/umd"
+        const ffmpeg = ffmpegRef.current
+        if (ffmpeg.loaded) return
+
+        ffmpeg.on("log", ({ message }) => {
+            log(message)
+        })
+
+        log("Loading ffmpeg...")
+
+        await ffmpeg.load({
+            coreURL: await toBlobURL(
+                `${baseURL}/ffmpeg-core.js`,
+                "text/javascript",
+            ),
+            wasmURL: await toBlobURL(
+                `${baseURL}/ffmpeg-core.wasm`,
+                "application/wasm",
+            ),
+        })
+    }
+
+    const fromHexString = (hexString: string) =>
+        Uint8Array.from(hexString.match(/.{1,2}/g)!.map((byte: string) => parseInt(byte, 16)))
+
+    const applyHeader = (file: Uint8Array): Uint8Array => {
+        const headerStart = fromHexString("564D5F534F554E44FFFFFFFF015858585800585858585858")
+        const fileWithHeader = new Uint8Array(headerStart.length + file.length + 4)
+
+        fileWithHeader.set(headerStart, 0)
+
+        const fileSize = file.byteLength
+        fileWithHeader.set([fileSize & 0xFF, (fileSize >> 8) & 0xFF, (fileSize >> 16) & 0xFF, (fileSize >> 24) & 0xFF], headerStart.length)
+
+        fileWithHeader.set(file, headerStart.length + 4)
+        return fileWithHeader
+    }
+
+    const startConversion = async () => {
+        if (converting) return
+        setConverting(true)
+
+        await loadFfmpeg()
+        const ffmpeg = ffmpegRef.current
+
+        log("Loading file into memory...")
+        await ffmpeg.writeFile(selectedFile.name, await fetchFile(selectedFile))
+
+        log("Converting file...")
+        const ffmpegArgs = "-acodec pcm_s16le -ac 1 -ar 22050 -map_metadata -1 -fflags +bitexact".split(" ")
```

Comment:
Would be nice if we can configure the sample rate, would allow us to use way more sounds

Reply:
Yeah, I think auto-selecting sample rate would actually probably be best. Since the size of the file scales linearly with sample rate and duration, we can figure out the highest sample rate that would still be under the 400KB limit.

Reply:
> Yeah, I think auto-selecting sample rate would actually probably be best. Since the size of the file scales linearly with sample rate and duration, we can figure out the highest sample rate that would still be under the 400KB limit.

Oh yeah I didn't really think about that, making it automatically would be best.